cpp/ql/src/change-notes/2024-12-18-return-stack-allocated-memory.md

```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* The "Returning stack-allocated memory" query (`cpp/return-stack-allocated-memory`) no longer produces results if there is an extraction error in the type of the function.
```

Comment:
This is not accurate. It no longer produces a result when there's an `ErroneousType` anywhere on the potential must-flow path (but that is too much detail for a change note). So it might be that the return type of the function is known, but that some critical type within the function is not.

Something along the following lines (modifying the example from the QL file):
```cpp
   int* test() {
      UNKNOWN_TYPE s;
      return &s.x; // BAD: &s.x is an address of a variable on the stack.
   }
```